src/utils/getFilteredPeople.ts

```diff
@@ -0,0 +1,61 @@
+import { Person } from '../types';
+
+export const getFilteredTodos = (
+  people: Person[],
+  query: string,
+  paramSex: string,
+  centuries: string[],
+  sortParam: string,
+  isReversed: boolean,
+): Person[] => {
+  let filteredPeople = [...people];
+
+  if (paramSex) {
+    filteredPeople = filteredPeople.filter(person => (
+      person.sex === paramSex
+    ));
+  }
+
+  if (centuries.length > 0) {
+    filteredPeople = filteredPeople.filter(person => {
+      const isCentury = centuries.some(century => (
+        century === Math.ceil(person.born / 100).toString()
+      ));
+
+      return isCentury;
+    });
+  }
+
+  if (query) {
+    const lowerCaseQuery = query.trim().toLowerCase();
+
+    filteredPeople = filteredPeople.filter(person => {
+      const lowerName = person.name.toLowerCase();
+      const lowerMotherName = person.motherName?.toLowerCase();
+      const lowerFatherName = person.fatherName?.toLowerCase();
+
+      return lowerName.includes(lowerCaseQuery)
+        || lowerMotherName?.includes(lowerCaseQuery)
+        || lowerFatherName?.includes(lowerCaseQuery);
+    });
+  }
+
+  filteredPeople.sort((personA, personB) => {
```

Comment:
I suppose, that better use condition on sort parameter, to avoid redundunt check